src/main/java/com/codeborne/selenide/Condition.java

```diff
@@ -771,4 +779,166 @@ public String getName() {
+   * is satisfied to the {@link LocalDateCondition#DEFAULT_PATTERN default pattern}
+   * and is equal to expected date
+   *
+   * <p>Sample: {@code $("input").shouldHave(date(expectedLocalDate));}</p>
+   *
+   * @param expectedDate expected value of "value" attribute
+   *
+   * @see Condition#date(LocalDate, String)
+   */
+  @CheckReturnValue
+  @Nonnull
+  public static Condition date(LocalDate expectedDate) {
+    return new LocalDateCondition(expectedDate, LocalDateCondition.DEFAULT_PATTERN);
+  }
+
+  /**
+   * Assert that element contains "value" attribute with date value that is satisfied to the pattern
+   *
+   * <p>Sample: {@code $("input").shouldHave(dateFormat("dd/MM/yyyy"));}</p>
+   *
+   * @param pattern date format of value of "value" attribute
+   */
+  @CheckReturnValue
+  @Nonnull
+  public static Condition dateFormat(String pattern) {
+    return new LocalDateFormatCondition(pattern);
+  }
+
+  /**
+   * Assert that element contains "value" attribute with date value that is satisfied to the pattern
+   * <br>
+   * and is inside interval {@code [startDate, endDate] (inclusive)}
+   *
+   * <p>Sample: {@code $("input").shouldHave(dateBetween(expectedStartDate, expectedEndDate, "dd/MM/yyyy"));}</p>
+   *
+   * @param startDate start of the interval (inclusive)
+   * @param endDate end of the interval (inclusive)
+   * @param pattern date format of value of "value" attribute
+   */
+  @CheckReturnValue
+  @Nonnull
+  public static Condition dateBetween(LocalDate startDate, LocalDate endDate, String pattern) {
+    return new LocalDateBetweenCondition(startDate, endDate, pattern);
+  }
+
+  /**
+   * Assert that element contains "value" attribute with date value that
+   * is satisfied to the {@link LocalDateCondition#DEFAULT_PATTERN default pattern}
+   * <br>
+   * and is inside interval {@code [startDate, endDate] (inclusive)}
+   *
+   * <p>Sample: {@code $("input").shouldHave(dateBetween(expectedStartDate, expectedEndDate));}</p>
+   *
+   * @param startDate start of the interval (inclusive)
+   * @param endDate end of the interval (inclusive)
+   *
+   * @see Condition#dateBetween(LocalDate, LocalDate, String)
+   */
+  @CheckReturnValue
+  @Nonnull
+  public static Condition dateBetween(LocalDate startDate, LocalDate endDate) {
+    return new LocalDateBetweenCondition(startDate, endDate, LocalDateCondition.DEFAULT_PATTERN);
+  }
+
+  /**
+   * Assert that element contains "value" attribute with datetime value
+   * that is satisfied to the pattern and is equal to expected datetime
+   *
+   * <p>Sample: {@code $("input").shouldHave(datetime(expectedLocalDateTime, "dd/MM/yyyy HH:mm:ss"));}</p>
+   *
+   * @param expectedDateTime expected value of "value" attribute
+   * @param pattern datetime format of value of "value" attribute
+   */
+  @CheckReturnValue
+  @Nonnull
+  public static Condition datetime(LocalDateTime expectedDateTime, String pattern) {
```

Comment:
Do you have integration tests that try to set/check time (or date+time)?

Try to add such inputs to `page_with_datetime_input.html`:

```html
      <input id="birthday" type="date" value="2022-10-11"/>
      <input id="birthtime" type="time" value="12:13:14"/>
      <input id="birthdate" type="datetime-local" value="2022-10-11 12:13:14"/>
```

Reply:
There are already  `page_with_date_inputs.html` with date and `page_with_datetime_inputs.html` with datetime

There is no ability to check time condition for now...)

Or did I misunderstand your comment?